src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -0,0 +1,254 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/CubicScale.hpp"
+
+#include <array>
+#include <functional>
+#include <ostream>
+#include <pup.h>
+#include <pup_stl.h>
+#include <vector>
+
+#include "ControlSystem/FunctionOfTime.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/RootFinding/CubicEquation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace CoordMapsTimeDependent {
+
+CubicScale::CubicScale(const double outer_boundary)
+    : outer_boundary_(outer_boundary) {}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::operator()(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  const auto a_of_t = map_list.at(f_of_t_a_).func(time)[0][0];
+  const auto b_of_t = map_list.at(f_of_t_b_).func(time)[0][0];
+  return {{source_coords[0] *
+           (a_of_t +
+            (b_of_t - a_of_t) * square(source_coords[0] / outer_boundary_))}};
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::inverse(
+    const std::array<T, 1>& target_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  // the original coordinates are found by solving for the roots
+  // of (b-a)/X^2*\xi^3 + a*\xi - x = 0, where a and b are the FunctionsOfTime,
+  // X is the outer_boundary, and x represents the mapped coordinates
+  const auto a_of_t = map_list.at(f_of_t_a_).func(time)[0][0];
+  const auto b_of_t = map_list.at(f_of_t_b_).func(time)[0][0];
+  const double cubic_coef_a = (b_of_t - a_of_t) / square(outer_boundary_);
+  const auto roots =
+      real_cubic_roots(cubic_coef_a, 0.0, a_of_t, -target_coords[0]);
+
+  // The map is invertible for all r<=outer_boundary as long as 0<a<3b/2.
+  if (b_of_t <= 2.0 / 3.0 * a_of_t) {
+    ERROR("The map is only invertible if expansion_b < expansion_a*2/3, however"
+          << " expansion_b = " << b_of_t << " and expansion_a = " << a_of_t
+          << " does not satisfy this criterion.");
+  }
+
+  std::vector<double> good_roots;
+  if (roots.size() > 1) {
```

Comment:
do something to the extent of `if roots.size ==1 {return root}` that way it is easier to see the branches